electrum/json_db.py

```diff
@@ -268,17 +347,32 @@ def _convert_value(self, path, key, v):
                 v = constructor(v)
         return v
 
-    def write(self, storage: 'WalletStorage'):
-        with self.lock:
-            self._write(storage)
+    @locked
+    def write(self):
+        if self.storage.file_exists() and not self.storage.is_encrypted():
+            self._append_pending_changes()
+        else:
+            self._write()
+
+    @locked
+    def _append_pending_changes(self):
+        if threading.currentThread().isDaemon():
```

Comment:
these methods are deprecated
```suggestion
        if threading.current_thread().daemon:
```